tesseract_command_language/include/tesseract_command_language/composite_instruction.h

```diff
@@ -65,6 +65,24 @@ class CompositeInstruction
   void setProfile(const std::string& profile);
   const std::string& getProfile() const;
 
+  /** @brief Add a profile object that will override the named profile for a specific task
+   * @param task_name Name of the task to which this profile will be applied
+   * @param profile ConstPtr to profile of appropriate type for the given task  */
+  void addProfileOverride(const std::string& task_name, std::shared_ptr<const void> profile);
```

Comment:
Should we use the profile dictionary here for storing the profiles?

Reply:
Probably. I forgot that existed. We'll have to move it into command language. Do you want me to just add a ProfileDictionary::Ptr as a public member?

Reply:
I believe it may be generic enough to go in tesseract_common?

> Do you want me to just add a ProfileDictionary::Ptr as a public member?

I think that should be fine. It may be worth checking the cpp core guidelines.

Reply:
I am making it a public ProfileDictionary::Ptr as indicated by [C131](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c131-avoid-trivial-getters-and-setters).

Also, I moved it into tesseract_command_language rather than tesseract_common since I figured that might be more likely to change and it is unlikely to be used in tesseeract core.